src/js/workshop-comments.js

```diff
@@ -0,0 +1,144 @@
+	if ((name === "") & (comment === "")) {
+		document.querySelectorAll(".required").forEach(element => {
+			applyVisibility(element, true, "inline");
+		});
+		disableFields(false);
+		return;
+	}
+
+	if (name === "") {
+		applyVisibility(document.querySelector(".required-name"), true, "inline");
+		disableFields(false);
+		return;
+	}
+
+	if (comment === "") {
+		applyVisibility(document.querySelector(".required-comment"), true, "inline");
+		disableFields(false);
+		return;
+	}
+
+	const XHR = new XMLHttpRequest();
+
+	const commentForm = document.getElementById("comment-form");
+	// Bind the FormData object and the form element
+	const commentFormData = new FormData( commentForm );
+
+	// submit and add comment to top of comment section of page on successful data submission
+	XHR.addEventListener( "load", function() {
+		if (XHR.status === 200) {
+			setSubmittedComment();
+			// Clear Text Fields
+			commentForm.reset();
+		} else {
+			// Show failed comment submission message
+			submittedCommentMessage.setAttribute("style", "display: block;");
+		}
+		disableFields(false);
+	});
+
+	// Show failure message in case of error
+	XHR.addEventListener( "error", function() {
+		// Show failed comment submission message
+		submittedCommentMessage.setAttribute("style", "display: block;");
+		disableFields(false);
+	} );
+
+	// Set up our request
+	XHR.open( "POST", window.location.origin + "/api/comments" );
+
+	// The data sent is what the user provided in the form
+	const jsonFormData = JSON.stringify(Object.fromEntries(commentFormData));
+	XHR.setRequestHeader("Content-Type", "application/json");
+	XHR.send(jsonFormData);
+});
+
+/*
+ * Hide/show required and success messages
+ * @param {DOMElement} DOMselector - The DOM Element to parse.
+ * @param {Boolean} toDisplay - value is true to show and false to hide
+ * @param {String}[Optional] displayType - CSS display value to apply
+ */
+function applyVisibility(DOMselector, toShow, displayType = "none") {
+	toShow ? DOMselector.setAttribute("style", `display: ${displayType};`) : DOMselector.setAttribute("style", "display: none;");
+}
+
+/*
+ * Add new comment to the given workshop's page temporarily.
+ */
+function setSubmittedComment() {
+	// Add comment to the top of the comments section of workshop page
+	const newCommentDiv = document.createElement("div");
+	newCommentDiv.innerHTML = "<article class='comment submitted-comment'><p><span class='comment-name'></span> | <span class='comment-date'></span></p><p class='comment-text'></p></article>";
+	const commentsDiv = document.getElementsByClassName("comments")[0];
+	commentsDiv.prepend(newCommentDiv.firstChild);
+
+	const submittedCommentDiv = commentsDiv.firstChild;
+	const submittedDate = submittedCommentDiv.querySelector(".comment-date");
+	const submittedName = submittedCommentDiv.querySelector(".comment-name");
+	const submittedComment = submittedCommentDiv.querySelector(".comment-text");
+
+	const dateObject = new Date();
```

Comment:
line 112-116 is a copy of `convertDate()` in the utils.js? Please directly call that function rather than a copy/paste. Thanks.